src/Avalonia.Controls/Button.cs

```diff
@@ -351,6 +375,28 @@ private static void IsDefaultChanged(AvaloniaPropertyChangedEventArgs e)
             }
         }
 
+        /// <summary>
+        /// Called when the <see cref="IsCancel"/> property changes.
+        /// </summary>
+        /// <param name="e">The event args.</param>
+        private static void IsCancelChanged(AvaloniaPropertyChangedEventArgs e)
+        {
+            var button = e.Sender as Button;
+            var isCancel = (bool)e.NewValue;
+
+            if (button?.VisualRoot is IInputElement inputRoot)
+            {
+                if (isCancel)
```

Comment:
Sorry, that I'm writing it here - it concerns and IsDefault property too. This just question to all. 
In wpf i'm using next pattern:
```
if ((bool) e.OldValue)
    Unsubscribe();
if ((bool) e.NewValue)
    Subscribe();
```
This can save from situation when e.OldValue==e.NewValue==true. I think you will get two subscriptions in this case. This is very sad situation.

I found just one situation how this can happen - this is call ```RaisePropertyChanged(IsDefaultProperty, IsDefault, IsDefault);``` directly. This is not look like real situation, but why not fix potential problem?

Reply:
I created separate PR for my question #2916
If it possible I think my question should not block approve of this PR.